truffleruby/src/main/java/org/truffleruby/core/thread/ThreadManager.java

```diff
@@ -389,4 +421,20 @@ public synchronized void accept(DynamicObject thread, Node currentNode) {
         }
     }
 
+    @TruffleBoundary
+    public void interrupt(Thread thread) {
+        final UnblockingAction action = unblockingActions.get(thread);
+
+        if (action != null) {
+            action.unblock();
+            unblockingActions.remove(thread);
```

Comment:
Okay. I originally did it that way so it wouldn't be removed until the unblocking action exited. But I suppose it doesn't matter much here.